capz/run-capz-e2e.sh

```diff
@@ -128,7 +127,13 @@ create_cluster(){
     if [[ ! "$SKIP_CREATE" == "true" ]]; then
         # create cluster
         log "starting to create cluster"
-        az extension add -y --upgrade --source "$CAPI_EXTENSION_SOURCE" || true
+
+        if [[ -z "$(command -v clusterctl)" ]]; then
+            log "install clusterctl"
+            curl -L https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.7.2/clusterctl-linux-amd64 -o /tmp/clusterctl
```

Comment:
Let's move the version of CAPI into a const at the top so it's easier to manage maintaining it and more conspicuous (and we upgrade it often)

Reply:
done